From 34d9a490373017a54a9f652d0f742254f946579c Mon Sep 17 00:00:00 2001
From: "Leonardo G. Trombetta" <lgtrombetta@gmx.com>
Date: Mon, 13 Mar 2023 17:45:19 +0100
Subject: [PATCH 52/90] arm64: dts: rockchip: rk3399-pinephone-pro: Add
 mount-matrix for magnetometer

After experimenting with the magnetometer on the Pinephone Pro I found out the
mount-matrix currently reported by the AF8133J driver is just the identity
matrix (no rotation), which is wrong. It would seem to me that there is
currently no `mount-matrix` info and that's why the driver reports a trivial
answer. As far as I can tell from my tests, a 90 degree clockwise rotation in
the XY plane would align it properly.

Signed-off-by: Leonardo G. Trombetta <lgtrombetta@gmx.com>
Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 927b481bd588..fd1c04c3868a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -978,6 +978,11 @@ af8133j: compass@1c {
 		reset-gpios = <&gpio1 RK_PA1 GPIO_ACTIVE_LOW>;
 		avdd-supply = <&vcc_3v0>;
 		dvdd-supply = <&vcc_1v8>;
+		
+		mount-matrix =
+			"0", "1", "0",
+			"-1", "0", "0",
+			"0", "0", "1";
 	};
 };
 
-- 
2.55.0

